torch/csrc/autograd/record_function.h

```diff
@@ -26,31 +28,79 @@ struct TORCH_API StringView {
   const char* str_ptr_;
 };
 
-using GetPackedInputsCallback = std::function<std::vector<c10::IValue>()>;
+enum class RecordFunctionState {
+  NO_CALLBACKS, // no callbacks registered, no need to use RecordFunction
```

Comment:
I still don't understand why we need this state in addition to the callback_needs_input flag.